v1/related.yaml

```diff
@@ -0,0 +1,133 @@
+swagger: '2.0'
+info:
+  version: '1.0.0-beta'
+  title: MediaWiki Related Pages API
+  description: Related pages API
+  termsofservice: https://github.com/wikimedia/restbase#restbase
+  contact:
+    name: Services
+    email: services@lists.wikimedia.org
+    url: https://www.mediawiki.org/wiki/Services
+  license:
+    name: Apache licence, v2
+    url: https://www.apache.org/licenses/LICENSE-2.0
+paths:
+  /related/{title}:
+    get:
+      tags:
+        - Page content
+      summary: Get pages related to the given title
+      description: |
+        Returns summaries for 5 pages related to the given page. Summaries include
+        page title, namespace and id along with short text description of the page
+        and a thumbnail.
+
+        Stability: [experimental](https://www.mediawiki.org/wiki/API_versioning#Experimental)
+      produces:
+        - application/json
+      parameters:
+        - name: title
+          in: path
+          description: The page title.
+          type: string
+          required: true
+      responses:
+        '200':
+          description: The related pages
+          schema:
+            $ref: '#/definitions/related'
+        '404':
+          description: Unknown page title
+          schema:
+            $ref: '#/definitions/problem'
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/problem'
+
+      x-request-handler:
+        # Storage miss. Call the Action API to get the textextract.
+        - api_request:
+            request:
+              method: 'post'
+              uri: /{domain}/sys/action/query
+              body:
+                format: 'json'
+                prop: 'pageterms|pageimages|pageprops'
+                ppprop: 'mainpage|disambiguation'
+                wbptterms: 'description'
+                generator: 'search'
+                gsrsearch: 'morelike:{request.params.title}'
+                gsrnamespace: 0
+                gsrwhat: 'text'
+                gsrinfo: ''
+                gsrprop: 'redirecttitle'
+                gsrlimit: 5
+                piprop: 'thumbnail'
+                pithumbsize: 640
```

Comment:
This is the old issue of not having an official API that lets clients select the thumb size. We are pushing for progress on https://phabricator.wikimedia.org/T66214, but in the meantime we'll have to work around this. Options:

1) Clients rewrite /\/320(px-[^\/])$/ to `/<desiredWidth>px-....jpg`. To accurately calculate resulting sizes, original dimensions or a very accurate aspect ratio should be provided in the response.
2) We provide several size options in the standard response.
3) Fragment on thumb size. While easy, this will seriously limit our hit rates, which is bad for latency and backend load.